contactql/parser.go

```diff
@@ -185,27 +185,31 @@ func (c *Condition) validate(env envs.Environment, resolver Resolver) error {
 			if err != nil {
 				return NewQueryError(ErrInvalidDate, "can't convert '%s' to a date", c.value).withExtra("value", c.value)
 			}
+		}
 
-		} else if c.propKey == AttributeGroup && resolver != nil {
-			group := c.ValueAsGroup(resolver)
-			if group == nil {
-				return NewQueryError(ErrInvalidGroup, "'%s' is not a valid group name", c.value).withExtra("value", c.value)
-			}
-		} else if (c.propKey == AttributeFlow || c.propKey == AttributeHistory) && resolver != nil {
-			flow := c.ValueAsFlow(resolver)
-			if flow == nil {
-				return NewQueryError(ErrInvalidFlow, "'%s' is not a valid flow name", c.value).withExtra("value", c.value)
-			}
-		} else if c.propKey == AttributeStatus {
-			val := strings.ToLower(c.value)
-			if val != "active" && val != "blocked" && val != "stopped" && val != "archived" {
-				return NewQueryError(ErrInvalidStatus, "'%s' is not a valid contact status", c.value).withExtra("value", c.value)
-			}
-		} else if c.propKey == AttributeLanguage {
-			if c.value != "" {
-				_, err := i18n.ParseLanguage(c.value)
-				if err != nil {
-					return NewQueryError(ErrInvalidLanguage, "'%s' is not a valid language code", c.value).withExtra("value", c.value)
+		// for some text attributes, do some additional validation
+		if c.propType == PropertyTypeAttribute {
```

Comment:
little bit of reorg here so that if you have a field called `language`.. we don't enforce that the value is valid language code etc